src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -736,13 +737,20 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
     protected function getContainerLoader(ContainerInterface $container)
     {
         $locator = new FileLocator($this);
-        $resolver = new LoaderResolver(array(
+        $loaders = array(
             new XmlFileLoader($container, $locator),
             new YamlFileLoader($container, $locator),
             new IniFileLoader($container, $locator),
             new PhpFileLoader($container, $locator),
             new ClosureLoader($container),
-        ));
+        );
+
+        // DirectoryLoader only appears in DI 2.7
+        if (class_exists('Symfony\\Component\\DependencyInjection\\Loader\\DirectoryLoader')) {
+            $loaders[] = new DirectoryLoader($container, $locator);
+        }
+
+        $resolver = new LoaderResolver($loaders);
```

Comment:
I am wondering why the default sets of loaders is written in `HttpKernel`, shouldn’t there be something like `Symfony\Component\DependencyInjection\Loader\LoaderResolver` that would contain the set of default loaders? This is out of scope of this PR though.